File: app/src/main/java/com/alamodrafthouse/injection/module/NetworkModule.java
package com.alamodrafthouse.injection.module;

import com.alamodrafthouse.BuildConfig;
import com.alamodrafthouse.data.DataManager;
import com.alamodrafthouse.data.services.ApiService;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by Shekar on 3/3/17.
 */

@Module public class NetworkModule {

  @Provides @Singleton OkHttpClient provideOkHttpClient() {
    OkHttpClient.Builder httpBuilder = new OkHttpClient.Builder();
    if (BuildConfig.DEBUG) {
      HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor();
      httpLoggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
      httpBuilder.interceptors().add(httpLoggingInterceptor);
    }
    return httpBuilder.build();
  }

  @Provides @Singleton Retrofit provideRestAdapter(OkHttpClient okHttpClient) {
    return new Retrofit.Builder().baseUrl(BuildConfig.BASE_URL)
        .addConverterFactory(GsonConverterFactory.create())
        .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
        .client(okHttpClient)
        .build();
  }

  @Provides @Singleton ApiService provideApiService(Retrofit restAdapter) {
    return restAdapter.create(ApiService.class);
  }

  @Provides @Singleton DataManager getDataManager(ApiService apiService) {
    return new DataManager(apiService);
  }
}

File: app/src/main/java/com/mutualmobile/praxis/injection/module/NetworkModule.kt
package com.mutualmobile.praxis.injection.module

import com.mutualmobile.praxis.AppConstants
import com.mutualmobile.praxis.data.remote.JokeApiService
import com.mutualmobile.praxis.data.remote.RetrofitHelper
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import javax.inject.Singleton

/**
 * Created by Vipul Asri on 13/01/21.
 */

@Module
@InstallIn(SingletonComponent::class)
object NetworkModule {

  @Provides
  @Singleton
  @JvmStatic
  fun provideHttpClient(): OkHttpClient {
    return RetrofitHelper.createOkHttpClient()
  }

  @Provides
  @Singleton
  @JvmStatic
  fun provideRetrofit(
    okHttpClient: OkHttpClient
  ): Retrofit {
    return RetrofitHelper.createRetrofitClient(okHttpClient, AppConstants.BASE_URL)
  }

  @Provides
  @Singleton
  @JvmStatic
  fun provideJokesApiService(retrofit: Retrofit): JokeApiService {
    return JokeApiService.createRetrofitService(retrofit)
  }

}